sympy/integrals/manualintegrate.py

```diff
@@ -687,11 +689,10 @@ def trig_substitution_rule(integral):
     if matches:
         for expr in matches:
             match = expr.match(a + b*symbol**2)
-            a = match[a]
-            b = match[b]
-
-            a_positive = ((a.is_number and a > 0) or a.is_positive)
-            b_positive = ((b.is_number and b > 0) or b.is_positive)
+            if match:
+                a,b = match.get(a),match.get(b)
+            a_positive = (a != None) and (((a > 0 and a.is_number) or ((not isinstance(a, int)) and a.is_positive)))
```

Comment:
```
Should you return if there is no match?
```
return ```None``` is good for normal test case but take more complicate example if we return ```None``` then we miss to return multiple solution set.see [sympy/solvers/tests/test_ode.py](https://github.com/sympy/sympy/blob/master/sympy/solvers/tests/test_ode.py#L543)
```
What are the default values if a and/or b are not in the match?
```
both are ```None```.
```
If one is in match, is the other guaranteed to be there, too?
```
yes